pallets/staking/src/inflation.rs

```diff
@@ -44,18 +44,20 @@ where
     const MILLISECONDS_PER_YEAR: u64 = 1000 * 3600 * 24 * 36525 / 100;
 
     let portion = Perbill::from_rational(era_duration as u64, MILLISECONDS_PER_YEAR);
-    // Have fixed rewards kicked in?
-    if total_tokens >= max_inflated_issuance {
-        let payout = portion * non_inflated_yearly_reward;
-        // payout is always maximum.
-        return (payout.clone(), payout);
-    }
 
     let payout = portion * yearly_inflation.calculate_for_fraction_times_denominator(
         npos_token_staked,
         total_tokens.clone(),
     );
-    let maximum = portion * (yearly_inflation.maximum * total_tokens);
+    let maximum = portion * (yearly_inflation.maximum * total_tokens.clone());
+    // Have fixed rewards kicked in?
+    if total_tokens >= max_inflated_issuance {
+        let fixed_payout = portion * non_inflated_yearly_reward;
+        if fixed_payout <= payout {
+            // payout is always maximum.
+            return (fixed_payout.clone(), fixed_payout);
+        }
+    }
```

Comment:
Optional minor change.  The `maximum` value can be calculated after the `if` and doesn't need the clone.
```suggestion
    // Have fixed rewards kicked in?
    if total_tokens >= max_inflated_issuance {
        let fixed_payout = portion * non_inflated_yearly_reward;
        if fixed_payout <= payout {
            // payout is always maximum.
            return (fixed_payout.clone(), fixed_payout);
        }
    }
    let maximum = portion * (yearly_inflation.maximum * total_tokens);
```